golang/cosmos/ante/ante.go

```diff
@@ -45,9 +45,7 @@ func NewAnteHandler(opts HandlerOptions) (sdk.AnteHandler, error) {
 	anteDecorators := []sdk.AnteDecorator{
 		ante.NewSetUpContextDecorator(),
 		ante.NewExtensionOptionsDecorator(nil), // reject all extensions
-		// former ante.NewMempoolFeeDecorator()
-		// replaced as in https://github.com/provenance-io/provenance/pull/1016
-		ante.NewDeductFeeDecorator(opts.AccountKeeper, opts.BankKeeper, opts.FeegrantKeeper, nil),
+		NewMempoolFeeDecorator(),
```

Comment:
Reviewing https://github.com/cosmos/cosmos-sdk/commit/d416ee86b6a000e564b88affb8d1c82b3ce72034 more closely, I think we _can_ drop MempoolFeeDecorator. As documented in upstream API Breaking Changes,
> The `MempoolFeeDecorator` has been removed. Instead, the `DeductFeeDecorator` takes a new argument of type `TxFeeChecker`, to define custom fee models. If `nil` is passed to this `TxFeeChecker` argument, then it will default to `checkTxFeeWithValidatorMinGasPrices`, which is the exact same behavior as the old `MempoolFeeDecorator` (i.e. checking fees against validator's own min gas price).

So checking fees against validator-local minimums still happens, it just got moved to a `checkTxFeeWithValidatorMinGasPrices` function in x/auth/ante/validator_tx_fee.go that is the default value of DeductFeeDecorator `txFeeChecker`. I think we should copy over x/auth/ante/validator_tx_fee.go to our golang/cosmos/ante/ and update our x/auth/ante/fee.go to bring in the changes of upstream cosmos-sdk x/auth/ante/fee.go.

```suggestion
```

Reply:
Yes, as noted above the updated `DeductFeeDecorator` appears to subsume `MempoolFeeDecorator` instead of being a 1:1 replacement for it.